### PR TITLE
[forge] Add state sync catching up stress tests

### PR DESCRIPTION
**Pull request by igor-aptos**
### Description

Adds two tests, where we send constant TPS, and have nodes get stale, and see that they catchup.

(will separately send a PR to add a check that they participate in the consensus)

- test 1 - cut network on a node for some period of time
- test 2 - make subset of nodes execute blocks slower (by adding sleep in aptos_vm block metadata processing)

### Test Plan
run both on forge to confirm they pass

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4036)

### CODE REVIEW

File: aptos-move/framework/cached-packages/src/aptos_framework_sdk_builder.rs [hunk 1]
```diff
@@ -362,6 +369,17 @@ pub enum EntryFunctionCall {
         new_beneficiary: AccountAddress,
     },
 
+    VestingSetBeneficiaryResetter {
```
Comment (sitalkedia) [inline aptos-move/framework/cached-packages/src/aptos_framework_sdk_builder.rs:372]: This is unrelated change?

File: testsuite/forge-cli/src/main.rs [hunk 2]
```diff
@@ -186,6 +186,14 @@ fn main() -> Result<()> {
     let duration = Duration::from_secs(args.duration_secs as u64);
     let suite_name: &str = args.suite.as_ref();
 
+    let suite_name = if suite_name == "compat" {
```
Comment (sitalkedia) [inline testsuite/forge-cli/src/main.rs:189]: Please remember to revert this before committing.

File: testsuite/testcases/src/consensus_reliability_tests.rs [hunk 3]
```diff
@@ -136,56 +144,87 @@ impl NetworkLoadTest for ChangingWorkingQuorumTest {
...
                 }
             }))),
-            Box::new(move |cycle, _, _, _, cur, previous| {
+            Box::new(move |cycle, _, _, _, cycle_end, cycle_start| {
                 let down_indices = down_indices_f(cycle);
-                let prev_down_indices = if cycle > 0 { down_indices_f(cycle - 1) } else { HashSet::new() };
-                fn split(all: Vec<NodeState>, down_indices: &HashSet<usize>) -> (Vec<(usize, NodeState)>, Vec<NodeState>) {
-                    let (down, active): (Vec<_>, Vec<_>) = all.into_iter().enumerate().partition(|(idx, _state)| down_indices.contains(idx));
-                    (down, active.into_iter().map(|(_idx, state)| state).collect())
+                let recently_down_indices = if cycle > 0 { down_indices_f(cycle - 1) } else { HashSet::new() };
```
Comment (sitalkedia) [inline testsuite/testcases/src/consensus_reliability_tests.rs:149]: Can you add some comments in this block explaining the high-level logic?

File: testsuite/testcases/src/consensus_reliability_tests.rs [hunk 4]
```diff
@@ -136,56 +144,87 @@ impl NetworkLoadTest for ChangingWorkingQuorumTest {
...
+                        bail!(
+                            "not enough individual progress on allowed lagging node ({}), only {} transactions, expected >= {} ({} TPS)",
+                            node_idx,
+                            transaction_progress,
+                            min_tps * check_period_s,
+                            min_tps,
+                        );
+                    }
+
+                    // recheck. state sync might not update rounds continuously ?
```
Comment (sitalkedia) [inline testsuite/testcases/src/consensus_reliability_tests.rs:204]: Please remove

File: testsuite/testcases/src/consensus_reliability_tests.rs [hunk 5]
```diff
@@ -136,56 +144,87 @@ impl NetworkLoadTest for ChangingWorkingQuorumTest {
...
-                        (check_period_s / 2),
+                    bail!(
+                        "not enough progress with active consensus, only {} epochs and {} rounds, expectd >= {}",
+                        epochs_progress,
+                        round_progress,
+                        check_period_s / 2,
                     );
                 }
 
+                // Make sure that allowed_lagging nodes are making progress
```
Comment (sitalkedia) [inline testsuite/testcases/src/consensus_reliability_tests.rs:190]: As a follow up, can we add a check to ensure the lagging nodes are eventually able to take part in consensus?

**Review by sitalkedia — APPROVED** [review]
Looks great % minor nits.

File: testsuite/forge.py [hunk 6]
```diff
@@ -1379,6 +1379,8 @@ def test(
     test_args: Optional[List[str]],
     test_suites: Tuple[str],
 ) -> None:
+    forge_enable_failpoints = "true"
```
Comment (bchocho) [inline testsuite/forge.py:1382]: What does this do? Does it enable failpoints everywhere?

File: testsuite/testcases/src/consensus_reliability_tests.rs [hunk 7]
```diff
@@ -89,23 +78,38 @@ impl NetworkLoadTest for ChangingWorkingQuorumTest {
...
+        let slow_allowed_lagging = if self.add_execution_delay {
             runtime.block_on(async {
                 let mut rng = rand::thread_rng();
-                for (name, validator) in &validators[num_always_healthy..num_validators] {
+                let mut slow_allowed_lagging = HashSet::new();
+                for (index, (name, validator)) in
+                    validators.iter().enumerate().skip(num_always_healthy)
+                {
                     let sleep_time = rng.gen_range(20, 500);
+                    if sleep_time > 100 {
```
Comment (bchocho) [inline testsuite/testcases/src/consensus_reliability_tests.rs:89]: This seems a bit arbitrary. Why do we have this threshold?

File: .github/workflows/continuous-e2e-state-sync-failures-catching-up-test.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  pull-requests: write
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 */3 * * *"
+
+jobs:
+  ### Please remember to use different namespace for different tests
+  # Performance test in an optimal setting
```
Comment (bchocho) [inline .github/workflows/continuous-e2e-state-sync-failures-catching-up-test.yaml:14]: Can remove?

File: .github/workflows/continuous-e2e-state-sync-slow-processing-catching-up-test.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  pull-requests: write
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 */3 * * *"
+
+jobs:
+  ### Please remember to use different namespace for different tests
+  # Performance test in an optimal setting
```
Comment (bchocho) [inline .github/workflows/continuous-e2e-state-sync-slow-processing-catching-up-test.yaml:14]: Can remove?

**Comment by igor-aptos** [comment]
@bchocho - addressed

**Review by bchocho — APPROVED** [review]
LGTM
